make compacting while truncating optional (#12857)
arangodb/arangodb: Documentation/DocuBlocks/Rest/Collections/put_api_collection_truncate.md

@RESTURLPARAM{collection-name,string,required}
The name of the collection.

@RESTQUERYPARAMETERS

@RESTQUERYPARAM{waitForSync,boolean,optional,}

> If *true* then the data is synchronized to disk before returning from a
> document create, update, replace or removal operation (default: *false*)

Reviewer: ```suggestion
If *true* then the data is synchronized to disk before returning from the
truncate operation (default: *false*)
```

@RESTQUERYPARAM{compact,boolean,optional,}
If *true* (default) then the storage engine is told to start a compaction

> in order to free up disk space. This can be resource intense. If the only 

Reviewer: ```suggestion
in order to free up disk space. This can be resource intensive. If the only 
```
